SerialPrograms/Source/PokemonSV/Inference/ItemPrinter/PokemonSV_ItemPrinterMaterialDetector.cpp

```diff
@@ -84,101 +85,31 @@ int16_t ItemPrinterMaterialDetector::read_number(
     Logger& logger, AsyncDispatcher& dispatcher,
     const ImageViewRGB32& screen, const ImageFloatBox& box
 ) const{
-    int16_t number_result_white_text = read_number_black_or_white_text(logger, dispatcher, screen, box, true);
-    if (number_result_white_text == -1){
-        // try looking for black text
-        return read_number_black_or_white_text(logger, dispatcher, screen, box, false);
-    }else{
-        return number_result_white_text;
-    }
-
-}
 
-int16_t ItemPrinterMaterialDetector::read_number_black_or_white_text(
-    Logger& logger, AsyncDispatcher& dispatcher,
-    const ImageViewRGB32& screen, const ImageFloatBox& box,
-    bool is_white_text
-) const{
     ImageViewRGB32 cropped = extract_box_reference(screen, box);
-
-    std::vector<BlackWhiteRgb32Range> filters; 
-    if (is_white_text){
-        filters = 
-        {
-            // white text filters
-            {0xff808080, 0xffffffff, true},
-            {0xff909090, 0xffffffff, true},
-            {0xffa0a0a0, 0xffffffff, true},
-            {0xffb0b0b0, 0xffffffff, true},
-            {0xffc0c0c0, 0xffffffff, true},
-            {0xffd0d0d0, 0xffffffff, true},
-            {0xffe0e0e0, 0xffffffff, true},
-            {0xfff0f0f0, 0xffffffff, true},
-        };
-    }else{
-        filters = 
-        {
-            // black text filters
-            // {0xff000000, 0xff101010, true},
-            // {0xff000000, 0xff202020, true},
-            {0xff000000, 0xff303030, true},
-            {0xff000000, 0xff404040, true},
-            {0xff000000, 0xff505050, true},
-            {0xff000000, 0xff606060, true},
-            {0xff000000, 0xff707070, true},
-            {0xff000000, 0xff808080, true},
-        };
-    }
-
-    std::vector<std::pair<ImageRGB32, size_t>> filtered = to_blackwhite_rgb32_range(
+    ImageRGB32 filtered = to_blackwhite_rgb32_range(
         cropped,
-        filters
+        0xff000000, 0xff808080,
+        true
     );
+    // filtered.save("DebugDumps/test-one-filter-1.png");
+    bool is_dark_text_light_background = image_stats(filtered).average.sum() > 500;
```

Comment:
I'm not sure if this is the best way to check if the text is dark vs light. Happy to hear any suggestions.

Reply:
500 feels too high. IIRC, I've had issues detecting whites in the past with 500 on my "bad" capture card.

What are the actual values for dark vs. light? Maybe pick the mid-point?

Reply:
Dark text light background is usually >600. Light text dark background is usually <200.
I've changed it to 400.  
Also, I do apply a binary filter first, so that should hopefully help as well.